### DISCUSSION TITLE
Sensor Node in distributed deployment not deployed correctly

### DISCUSSION THREAD

**Discussion by 8bitjoe** [Security-Onion-Solutions/securityonion#11224] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

installation

### Installation Type

Distributed

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### Network Traffic Collection

span port

### Network Traffic Speeds

Less than 1Gbps

### Status

No, one or more services are failed (please provide detail below)

### Salt Status

Yes, there are salt failures (please provide detail below)

### Logs

Yes, there are additional clues in /opt/so/log/ (please provide detail below)

### Detail

I'm adding a sensor node to a brand new distributed deployment. Manager and search node installation went smoothly without hiccups, however when i add the sensor node, the installation and so-setup TUI script works great but the node never installs docker, so-status says it's not available, nothing seems to happen.
Seems to me the salt configuration is failing but I'm not sure why...
On the first few lines of the salt log there's a reference to a cached key on the Master node, not sure if this could be an indication of the issue, (i've tried to install this node twice now).
The minion salt logs on the sensor node is quite extensive so I'll paste a link to the first 7k lines and another one for the last 7k
start_log: https://pastebin.com/BD2K2sJ0
end_log: https://pastebin.com/bCaXFUBW

Let me know what additional logs i should include to help troubleshoot.

More info on the infrastructure:
the nodes are virtual machines on a proxmox server, the only difference as far as the VM configuration on the hypervisor is that I'm passing through a NIC card via PCI-E Passthrough to the sensor node. this NIC card receives a cable from a span port, i can verify with tcpdump that it is working on the node.
The only other file found in /opt/so is an install.txt file with the following content:
MAINIP=X.X.2.9
MNIC=enp6s18
NODE_DESCRIPTION='Sensor Node for Security Onion Cluster'
ES_HEAP_SIZE=5298m
PATCHSCHEDULENAME=
INTERFACE=bond0
NODETYPE=SENSOR
CORECOUNT=5
LSHOSTNAME=so-sensor
LSHEAP=500m
CPUCORES=12
IDH_MGTRESTRICT=
IDH_SERVICES=

Administration interface is working and i can ssh into the node.

Even more information on the specs assigned to the VM:
Operating System Disk: 100G
NSM Disk: 1.5T
RAM: 16G
VCPU: 12 (Type Host)
BIOS: BIOS not uefi
Machine Type on Proxmox: Q35

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by 8bitjoe** [comment, marked as answer]
OK so i noticed that the shell script so-bpf-compile was missing from the sensor node looking at the salt logs, i copied it from the manager and now it seems that things have started working, however i will monitor it for a bit.
Please keep the issue open for a day or two. Thank you

**Comment by dougburks** [comment]
Were you running any third-party software that may have prevented so-bpf-compile from being written to disk or perhaps quarantined it?

**Reply by 8bitjoe** [reply]
nope, fresh installation with the official ISO. no customization whatsoever. No EDR, or AV or Third party software running on either the VM or the Proxmox host.
Everything has been running smoothly since so the issue can be closed, will leave the answer here just in case it helps someone.
